test/performance/odp_timer_perf.c

```diff
@@ -17,6 +17,7 @@
 
```

Comment:
The original intention is to add a case which can measure the timer expire/scan performance. Then we can optimize and measure the timer algorithm better.
As timer expire/scan function is not exposed to the outside, I can only implement profiling in linux-gen and use odp_schedule to call these functions and control expire numbers.
Maybe we can rename the "scan mode" to a more generic one?

Reply:
Wouldn't the existing 0-mode ("measure odp_schedule() overhead when using timers") already suffice for the scan profiling?

Reply:
Think existing 0-mode would do many other operations and it may not hit timer_pool_scan. Compare the below results for scan and schedule, we can find that odp_schedule() is much heavier that scan. Existing sched results can not measure the timer algorithm directly.

Maybe we can integrate scan mode and sched mode to make scan results as supplement to sched mode.
What's your opinion about this?

Scan results:
Inline timer profiling for thread 1:
  scan rounds:     2209
  ave scan nsec:   45.1
Inline timer profiling for thread 2:
  scan rounds:     2208
  ave scan nsec:   42.5

Schedule results:
TOTAL (2 workers)
  events:             10
  ave time:           1.10 sec
  ave rounds per sec: 23.98M
  num before:         0
  ave before:         0.0 nsec
  max before:         0 nsec
  num after:          10
  ave after:          3666524.0 nsec
  max after:          3922400 nsec

Reply:
Is the "Scan result" section above now with the new 2-mode?

Reply:
Nop, I enabled mode-0 to show the above "Scan Results" for comparison.

Reply:
OK. With the `odp_timer.c` changes one gets the impression that profiling and potential performance improvements are wanted for the `timer_pool_scan_inline()` function. In my local env, I get practically identical "ave scan nsec" regardless of the mode (0-mode or 2-mode), the "scan rounds" differ a lot with default values but that can be controlled with the period and resolution parameters.

Do "ave scan nsec" values vary a lot in your env depending on the mode?

Reply:
Just like yours, 'ave scan nsec' is almost the same for 0-mode and 2-mode in my env.
Think this is resonable. For both two modes, default timer numbers are the same and ave scan efficiency is positive correlated to timer numbers.

Reply:
OK, so no need for the new mode then?

Reply:
Agree with you, I would integrate the modes in next version.
Many thanks.